managed-pointer/cpp-20/D0567R1.md

```diff
@@ -229,94 +391,105 @@ auto gpuExecutor = gpuContext.executor();
+### 5.2. Implicit Sycnhronisation
 
-This could be done by having an additional concept which allows an execution context to also be a host device, making it capable of triggering synchronisation operations. This would allow any such device to construct a `managed_ptr` and allocate memory locally as well as allowing calls to the ``put()`` and ``get()``. In this case the synchronisation operations would be relative to the current host device. So, for example, a ``put()`` on device A to device B would be the equivalent to a ``get()`` on device B from device A. When it comes to memory reclamation for the remote device memory allocations, this would likely be handled by the host device that destroys the final reference to the managed memeory allocation.
+In a previous revision of this paper it was possible to trigger implicit sycnhronisation operations by simply passing a `managed_ptr` to a control structure such as `async`. However this has been removed due to limitations that may effect an implementation. As the current executors interface does not support variadic execution function parameters, implicit an implementation would be required to trigger sycnrhonisation operations on all control structures for `managed_ptr` captures. Firstly this may incur unwanted overhead and secondly this could potentially require static reflection for an optimal implementation. For these reasons this feature has been removed and instead kept as a potential future work that could be incorporated if proven to be implementable without creating overhead.
 
-### Additional Containers
+### 5.3. Additional Containers
 
-We may wish to extend this principle of a managed pointer to other containers that would be useful to share across heterogeneous and distributed systems such as vectors or arrays. This could be done by having containers such as ``managed_vector`` or ``managed_array`` that would have similar requirements to the standard containers of the same names in terms of storage and access yet would be extended to support access from remote devices as with the ``managed_ptr``.
+We may wish to extend this principle of a managed pointer to other containers that would be useful to share across heterogeneous and distributed systems such as vectors or arrays. This could be done by having containers such as `managed_vector` or `managed_array` that would have similar requirements to the standard containers of the same names in terms of storage and access yet would be extended to support access from remote devices as with the `managed_ptr`.
```

Comment:
Seems that you should be able to build containers on top of the managed_ptr , so no need for complicating the interface more